filesmanager/filesmanager.py

```diff
@@ -349,6 +355,93 @@ def delete_content(self, data, suffix=''):
             "status": "success",
         }
 
+    @XBlock.json_handler
+    def fill_directories(self, data, suffix=''):
+        """Fill the directories list with the content of the course assets.
+
+        This unorganized content will be added to the uncategorized directory, which is the first
+        directory in the directory list.
+
+        Returns: None.
+        """
+        self.prefill_directories()
+        return {
+            "status": "success",
+            "content": self.directories,
+        }
+
+    def initialize_uncategorized_directory(self):
+        """Initialize the uncategorized directory.
+
+        Returns: None.
+        """
+        self.directories = [
+            {
+                "name": "uncategorized",
+                "type": "directory",
+                "path": "uncategorized",
+                "metadata": {},
+                "children": [],
+            }
+        ]
+
+    def prefill_directories(self):
+        """Prefill the directories list with the content of the course assets.
+
+        This unorganized content will be added to the uncategorized directory, which is the first
+        directory in the list.
+
+        Returns: None.
+        """
+        uncategorized_directory = self.directories[0]
```

Comment:
Note to @johnvente: can we avoid the user moving a directory to the first position of the directory arrangement? So we always have the uncategorize directory first?

Reply:
Actually, no. I'm referring to the `unpublished` (or uncategorized) directory which lives in the root directory. Our root directory is the 1st directory level:
```

        [
        # This is root, parent of all new directories:
            {
                "name": "Folder 1",
                "type": "directory",
                "path": "Folder 1",
                "metadata": {
                    "id": ..,
                    ...
                },
                "children": [
                    ...
                ]
            }
        ]
```
The unpublished directory lives in the root directory at the first level. Should this behavior be different? Or we can work with it?

Reply:
You can read more about it [here](https://github.com/eduNEXT/xblock-filesmanager/pull/5#discussion_r1355279798). Let me know what you think!